Start tests with react (#1705)
cocotb/cocotb: documentation/source/newsfragments/1705.bugfix.rst

> Scheduling behavior in the beginning of a test, before the first ``await`` of a :class:`~cocotb.triggers.GPITrigger`, now matches the rest of the test.

Reviewer: ```suggestion
Scheduling behavior is now consistent before and after the first ``await`` of a :class:`~cocotb.triggers.GPITrigger`.
```
?

Author: Should we add bugfix info about `SIM_TEST_FAIL` too?

Reviewer: That's a good idea, I'm not sure how to have multiple newsfrags per PR in the same category though.

Author: I could break the `finish_test` changes into another PR and get that in first, or we can just use the combined newsfragment here.

Reviewer: [This](https://cocotb--1705.org.readthedocs.build/en/1705/release_notes.html#bugfixes) is what is currently looks like, we don't have 2 bullet points. I'll that up to @eric-wieser.

Reviewer 2: I think you can create a second fragment, `1705.bugfix.1.rst`
